src/generator/iter.rs

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /// Skips events until the next one that can be handled again.
+    pub fn skip(
+        &mut self, gen: &mut Generator<'a, impl Backend<'a>, impl Write>,
+    ) -> FatalResult<()> {
+        match self.last_kind {
+            // continue consuming until end
+            FeEventKind::Start => (),
+            // TODO: unsure if `End` might just be unrecoverable
+            FeEventKind::End
+            | FeEventKind::Text
+            | FeEventKind::Html
+            | FeEventKind::InlineHtml
+            | FeEventKind::Latex
+            | FeEventKind::FootnoteReference
+            | FeEventKind::BiberReferences
+            | FeEventKind::Url
+            | FeEventKind::InterLink
+            | FeEventKind::Include
+            | FeEventKind::Label
+            | FeEventKind::SoftBreak
+            | FeEventKind::HardBreak
+            | FeEventKind::TaskListMarker
+            | FeEventKind::Command
+            | FeEventKind::ResolveInclude => return Ok(()),
+        }
+        let mut depth = 0;
+        loop {
+            let evt = if let Some((peek, _, _)) = self.peek.take() {
+                peek
+            } else {
+                self.next(gen)?.unwrap().0
+            };
+            match evt {
+                Event::Start(_) => depth += 1,
+                Event::End(_) if depth > 0 => depth -= 1,
+                Event::End(_) => return Ok(()),
+                _ => {},
+            }
+        }
+    }
+
+    /// Converts an event, resolving any includes. If the include is handled, returns Ok(None).
+    /// If it fails, returns the original event.
+    fn convert_event(
+        &mut self, event: FeEvent<'a>, range: Range<usize>,
+        gen: &mut Generator<'a, impl Backend<'a>, impl Write>,
+    ) -> Result<Event<'a>> {
+        match event {
+            FeEvent::Include(image) => {
+                let include = gen.resolve(&image.dst, range.clone())?;
+                self.convert_include(include, Some(image), range, gen)
+            },
+            FeEvent::ResolveInclude(include) => {
+                let include = gen.resolve(&include, range.clone())?;
+                self.convert_include(include, None, range, gen)
+            },
+            e => Ok(e.into()),
+        }
+    }
+
+    fn convert_include(
+        &mut self, include: Include, image: Option<FeInclude<'a>>, range: Range<usize>,
+        gen: &mut Generator<'a, impl Backend<'a>, impl Write>,
+    ) -> Result<Event<'a>> {
+        match include {
+            Include::Command(command) => Ok(command.into()),
+            Include::Markdown(path, context) => {
+                let markdown = fs::read_to_string(&path).map_err(|err| {
+                    gen.diagnostics()
+                        .error("error reading markdown include file")
+                        .with_error_section(&range, "in this include")
+                        .error(format!("cause: {}", err))
```

Comment:
Include the resolved path: `info(format!("{}", &path))`. Several other tools omit these, making it very unclear why some `NotFound` actually occurs.. Also, this currently doesn't occur as it crashes during path resolution for an unfound file. What?

Reply:
> Also, this currently doesn't occur as it crashes during path resolution for an unfound file.

That should be fixed with #75. After that this diagnostic should be fine.